### PR TITLE
Aligning no-cache with force

### PR DESCRIPTION
**Pull request by nkolev92**
The force command broke the no-cache command
The goal of this PR to align it, so in case NoCache option is used, restore always forces (hence it hits the remote sources).

Fixes https://github.com/NuGet/Home/issues/5551

### CODE REVIEW

File: test/NuGet.Clients.FuncTests/NuGet.CommandLine.FuncTest/Caching/CachingTests.cs [hunk 1]
```diff
@@ -232,10 +242,55 @@ public class CachingTests
...
+        [Theory]
+        [InlineData(typeof(RestoreProjectJsonCommand), CachingType.Default, ServerType.V2, true, false)]
+        [InlineData(typeof(RestoreProjectJsonCommand), CachingType.Default, ServerType.V3, true, false)]
+        [InlineData(typeof(RestoreProjectJsonCommand), CachingType.NoCache, ServerType.V2, true, true)]
+        [InlineData(typeof(RestoreProjectJsonCommand), CachingType.NoCache, ServerType.V3, true, true)]
+        [InlineData(typeof(RestoreProjectJsonCommand), CachingType.DirectDownload, ServerType.V2, true, false)]
+        [InlineData(typeof(RestoreProjectJsonCommand), CachingType.DirectDownload, ServerType.V3, true, false)]
+        [InlineData(typeof(RestoreProjectJsonCommand), CachingType.NoCache | CachingType.DirectDownload, ServerType.V2, true, true)]
+        [InlineData(typeof(RestoreProjectJsonCommand), CachingType.NoCache | CachingType.DirectDownload, ServerType.V3, true, true)]
+        public async Task NuGetExe_Caching_DoesNotNoOp(Type type, CachingType caching, ServerType server, bool success, bool doesNotNoOp)
```
Comment (joelverhagen) [inline test/NuGet.Clients.FuncTests/NuGet.CommandLine.FuncTest/Caching/CachingTests.cs]: Can we invert the variable name and invert the booleans? `noOps` seems a bit more terse.

File: test/NuGet.Clients.FuncTests/NuGet.CommandLine.FuncTest/Caching/CachingValidationType.cs [hunk 2]
```diff
@@ -11,5 +11,6 @@ public enum CachingValidationType
         PackageFromSourceNotUsed,
         PackageFromGlobalPackagesFolderUsed,
         DirectDownloadFilesDoNotExist,
+        RestoreDoesNotNoOp
```
Comment (joelverhagen) [inline test/NuGet.Clients.FuncTests/NuGet.CommandLine.FuncTest/Caching/CachingValidationType.cs]: Nit end line in semicolon to make future diffs clearer

File: test/NuGet.Clients.FuncTests/NuGet.CommandLine.FuncTest/Caching/Tests/AllowsMissingPackageOnSourceTest.cs [hunk 3]
```diff
@@ -1,4 +1,4 @@
-﻿using System.Threading.Tasks;
+using System.Threading.Tasks;
```
Comment (joelverhagen) [inline test/NuGet.Clients.FuncTests/NuGet.CommandLine.FuncTest/Caching/Tests/AllowsMissingPackageOnSourceTest.cs]: copyright header?

File: test/NuGet.Clients.FuncTests/NuGet.CommandLine.FuncTest/Caching/Tests/UsesGlobalPackageFolderCopyOnEveryRunTest.cs [hunk 4]
```diff
@@ -0,0 +1,53 @@
+using System.Threading.Tasks;
+using NuGet.Test.Utility;
+
+namespace NuGet.CommandLine.Test.Caching
+{
+    public class UsesGlobalPackageFolderCopyOnEveryRunTest : ICachingTest
+    {
+        public string Description => "Uses the copy of the package from the global packages folder instead of the source on every iteration";
+
+        public int IterationCount => 3;
```
Comment (joelverhagen) [inline test/NuGet.Clients.FuncTests/NuGet.CommandLine.FuncTest/Caching/Tests/UsesGlobalPackageFolderCopyOnEveryRunTest.cs:13]: 👍

File: test/NuGet.Clients.FuncTests/NuGet.CommandLine.FuncTest/Caching/Tests/UsesGlobalPackageFolderCopyOnEveryRunTest.cs [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+        }
+
+        public CachingValidations Validate(CachingTestContext context, ICachingCommand command, CommandRunnerResult result)
+        {
+            var validations = new CachingValidations();
+
+            validations.Add(
+                CachingValidationType.CommandSucceeded,
+                result.Item1 == 0);
+
```
Comment (joelverhagen) [inline test/NuGet.Clients.FuncTests/NuGet.CommandLine.FuncTest/Caching/Tests/UsesGlobalPackageFolderCopyOnEveryRunTest.cs:34]: nit extra line

File: src/NuGet.Core/NuGet.Commands/RestoreCommand/RequestFactory/RestoreArgs.cs [hunk 6]
```diff
@@ -212,7 +212,7 @@ public void ApplyStandardProperties(RestoreRequest request)
                 request.ValidateRuntimeAssets = ValidateRuntimeAssets.Value;
             }
 
-            request.AllowNoOp = AllowNoOp;
+            request.AllowNoOp = request.CacheContext.NoCache ? false : AllowNoOp;
```
Comment (joelverhagen) [inline src/NuGet.Core/NuGet.Commands/RestoreCommand/RequestFactory/RestoreArgs.cs]: Seems weird to mention the `false` literal. `!request.CacheContext.NoCache && AllowNoOp`?

**Review by joelverhagen — APPROVED** [review]
Couple nits, but looks great.

**Comment by nkolev92** [comment]
Addressed the feedback.

**Review by emgarten — APPROVED** [review]

**Pull request merged**